### PR TITLE
RemovedAction should delegate isDumbAware 

### LINKED ISSUE #3788: RemovedAction prevents DumbAware actions from being triggered in Dumb Mode
We noticed that some actions, such as Build Project, were no longer available during indexing (AKA Dumb Mode) even in a non-bazel project. After some debugging I identified the cause was ReplaceActionHelper wrapped the action with RemovedAction which abstracted the original action from the DumbAware check. 

This can also easily be reproduced via Tools > Internal Actions > Enter Dumb Mode.

The fix seems to be simply to implement PossiblyDumbAware

### PR DESCRIPTION
**Pull request by brian-mcnamara**
# Checklist

- [ ] I have filed an issue about this change and discussed potential changes with the maintainers.
- [ ] I have received the approval from the maintainers to make this change.
- [x] This is not a stylistic, refactoring, or cleanup change.

Please note that the maintainers will not be reviewing this change until all checkboxes are ticked. See 
the [Contributions](https://github.com/bazelbuild/intellij#contributions) section in the README for more 
details.

# Discussion thread for this change

Issue number: Fixes #3788

# Description of this change

### CODE REVIEW

**Review by tpasternak — APPROVED** [review]
Hey @brian-mcnamara, thanks for this change, LGTM.

File: common/actions/src/com/google/idea/common/actions/ReplaceActionHelper.java [hunk 1]
```diff
@@ -67,7 +68,7 @@ public static void replaceAction(
   }
 
   /** Wraps an action and makes it conditionally invisible. */
-  private static class RemovedAction extends AnAction {
+  private static class RemovedAction extends AnAction implements PossiblyDumbAware {
```
Comment (jastice) [inline common/actions/src/com/google/idea/common/actions/ReplaceActionHelper.java:71]: The explicit implements here seems to be redundant.

**Review by jastice — APPROVED** [review]
Thanks!

File: common/actions/src/com/google/idea/common/actions/ReplaceActionHelper.java (hunk 1 above)
Comment (brian-mcnamara) [reply common/actions/src/com/google/idea/common/actions/ReplaceActionHelper.java]: You're absolutely right. Removed.

**Pull request merged**